tests/test_offchain.py

```diff
@@ -0,0 +1,91 @@
+import rapidjson
+from cryptoconditions import Fulfillment
+
+from pytest import raises, mark
+
+
+@mark.parametrize('operation,function,return_value', (
+    ('CREATE', 'prepare_create_transaction', 'create'),
+    ('TRANSFER', 'prepare_transfer_transaction', 'transfer'),
+))
+def test_prepare_transaction(operation, return_value, function, monkeypatch):
+    from bigchaindb_driver import offchain
+    from bigchaindb_driver.offchain import prepare_transaction
+
+    def mock(owners_before=None, owners_after=None,
+             inputs=None, asset=None, metadata=None):
+        return return_value
+    monkeypatch.setattr(offchain, function, mock)
+    assert prepare_transaction(operation=operation) == return_value
+
+
+def test_prepare_transaction_raises():
+    from bigchaindb_driver.offchain import prepare_transaction
+    from bigchaindb_driver.exceptions import BigchaindbException
+    with raises(BigchaindbException):
+        prepare_transaction(operation=None)
+
+
+@mark.parametrize('asset', (
+    None, {}, '', {'data': {'msg': 'Hello BigchainDB!'}},
+))
+def test_prepare_create_transaction(asset, alice_keypair):
+    from bigchaindb_driver.offchain import prepare_create_transaction
+    create_transaction = prepare_create_transaction(
+        owners_before=alice_keypair.vk, asset=asset)
+    assert 'id' in create_transaction
+    assert 'transaction' in create_transaction
+    assert 'version' in create_transaction
+    assert 'asset' in create_transaction['transaction']
+    assert 'conditions' in create_transaction['transaction']
+    assert 'fulfillments' in create_transaction['transaction']
+    assert 'metadata' in create_transaction['transaction']
+    assert 'operation' in create_transaction['transaction']
+    assert 'timestamp' in create_transaction['transaction']
+    assert create_transaction['transaction']['operation'] == 'CREATE'
+
+
+def test_prepare_transfer_transaction(alice_transaction,
+                                      bob_keypair,
+                                      alice_keypair):
+    from bigchaindb_driver.offchain import prepare_transfer_transaction
+    condition_index = 0
+    condition = alice_transaction['transaction']['conditions'][condition_index]
+    input_ = {
+        'fulfillment': condition['condition']['details'],
+        'input': {
+            'cid': condition_index,
+            'txid': alice_transaction['id'],
+        },
+        'owners_before': condition['owners_after']
+    }
+    asset = alice_transaction['transaction']['asset']
+    transfer_transaction = prepare_transfer_transaction(
+        inputs=input_, owners_after=bob_keypair.vk, asset=asset)
+    assert 'id' in transfer_transaction
+    assert 'transaction' in transfer_transaction
+    assert 'version' in transfer_transaction
+    assert 'asset' in transfer_transaction['transaction']
```

Comment:
Would be nice to double check if the asset generated for the transfer matches up with the input's.